src/GitHub.App/Services/UsageTracker.cs

```diff
@@ -0,0 +1,374 @@
+                        });
+                }
+                catch (Exception ex)
+                {
+                    // This should never happen, we have a catch in the observable
+                    // chain and se subscribe on the task pool but just to be extra
+                    // cautious let's deal with this scenario as well because this
+                    // method should never throw.
+                    log.Warn("Failed incrementing usage data counter", ex);
+                    return Observable.Empty<int>();
+                }
+            });
+        }
+
+        public void IncrementCloneCount()
+        {
+            IncrementCounter(GHCloneCountKey)
+                .Subscribe();
+        }
+
+        public void IncrementCreateCount()
+        {
+            IncrementCounter(GHCreateCountKey)
+                .Subscribe();
+        }
+
+        public void IncrementLaunchCount()
+        {
+            IncrementCounter(GHLaunchCountKeyDay)
+                .ContinueAfter(() => IncrementCounter(GHLaunchCountKeyWeek))
+                .ContinueAfter(() => IncrementCounter(GHLaunchCountKeyMonth))
+                .Subscribe();
+        }
+
+        public void IncrementPublishCount()
+        {
+            IncrementCounter(GHPublishCountKey)
+                .Subscribe();
+        }
+
+        public void IncrementOpenInGitHubCount()
+        {
+            IncrementCounter(GHOpenInGitHubCountKey)
+                .Subscribe();
+        }
+
+        public void IncrementLinkToGitHubCount()
+        {
+            IncrementCounter(GHLinkToGitHubCountKey)
+                .Subscribe();
+        }
+
+        public void IncrementCreateGistCount()
+        {
+            IncrementCounter(GHCreateGistCountKey)
+                .Subscribe();
+        }
+
+        public void IncrementUpstreamPullRequestCount()
+        {
+            IncrementCounter(GHUpstreamPullRequestCount)
+                .Subscribe();
+        }
+
+        IObservable<Unit> OptOut()
+        {
+            if (client != null)
+            {
+                return Observable.Defer(() =>
+                {
+                    log.Info("User has disabled sending anonymized usage statistics to GitHub");
+
+                    return ClearCounters(true, true)
+                        .ContinueAfter(() => client.SendOptOut());
+                });
+            }
+            else
+            {
+                return Observable.Return(Unit.Default);
+            }
+        }
+
+        IObservable<Unit> OptIn()
+        {
+            if (client != null)
+            {
+                return Observable.Defer(() =>
+                {
+                    log.Info("User has enabled sending anonymized usage statistics to GitHub");
+
+                    return client.SendOptIn();
+                });
+            }
+            else
+            {
+                return Observable.Return(Unit.Default);
+            }
```

Comment:
GHfW is not going to have any changes to this code, so it doesn't matter if it's identical or not (and it's not anymore anyway with the other PR coming in). I'd rather have consistency in the code base  and make this ours.

To be very honest, I'd rather have 

```
            return client != null
                ? Observable.Defer(() =>
                {
                    log.Info("User has enabled sending anonymized usage statistics to GitHub");
                    return client.SendOptIn();
                })
                : Observable.Return(Unit.Default);
```

because that's the form that we use the most in logic like this around the codebase, but that doesn't get on my nerves as much as those floating braces :wink: